common/src/main/java/com/wynntils/features/inventory/UnidentifiedItemIconFeature.java

```diff
@@ -55,41 +78,76 @@ public void onHotbarSlotRender(HotbarSlotRenderEvent.CountPre e) {
     }
 
     private void drawIcon(PoseStack poseStack, ItemStack itemStack, int slotX, int slotY, int z) {
-        Optional<GearBoxItem> gearBoxItemOpt = Models.Item.asWynnItem(itemStack, GearBoxItem.class);
-        if (gearBoxItemOpt.isEmpty()) return;
-
-        GearType gearType = gearBoxItemOpt.get().getGearType();
-
-        int textureX = TEXTURE_COORDS.get(gearType).a();
-        int textureY = TEXTURE_COORDS.get(gearType).b();
-        RenderUtils.drawTexturedRect(
-                poseStack,
-                Texture.GEAR_ICONS.resource(),
-                slotX + 2,
-                slotY + 2,
-                z,
-                12,
-                12,
-                textureX,
-                textureY + texture.get().getTextureYOffset(),
-                16,
-                16,
-                Texture.GEAR_ICONS.width(),
-                Texture.GEAR_ICONS.height());
+        Optional<WynnItem> wynnItemOpt = Models.Item.getWynnItem(itemStack);
+        if (wynnItemOpt.isEmpty()) return;
+
+        WynnItem wynnItem = wynnItemOpt.get();
+        if (wynnItem instanceof GearBoxItem box) {
+            texture.get().getIconRenderer().renderIcon(poseStack, slotX, slotY, z, box.getGearType());
+        } else if (showOnUnboxed.get() && wynnItem instanceof GearItem gear && gear.isUnidentified()) {
+            unboxedTexture.get().getIconRenderer().renderIcon(poseStack, slotX, slotY, z, gear.getGearType());
+        }
+    }
+
+    @FunctionalInterface
+    private interface IconRenderer {
+        void renderIcon(PoseStack poseStack, int x, int y, int z, GearType gearType);
+
+        static IconRenderer forSpriteSheet(Texture texture, int yOffset, int padding) {
+            int paddedDims = 16 - padding - padding;
+            return (poseStack, x, y, z, gearType) -> {
+                Pair<Integer, Integer> textureCoords = TEXTURE_COORDS.get(gearType);
+                RenderUtils.drawTexturedRect(
+                        poseStack,
+                        texture.resource(),
+                        x + padding,
+                        y + padding,
+                        z,
+                        paddedDims,
+                        paddedDims,
+                        textureCoords.a(),
+                        textureCoords.b() + yOffset,
+                        16,
+                        16,
+                        texture.width(),
+                        texture.height());
+            };
+        }
+
+        static IconRenderer forText(Function<GearType, StyledText> textMap) {
```

Comment:
@magicus the question mark is rendered here, as a font.

Reply:
I saw that later. Thanks.